### ISSUE TITLE
Make check_package_name optional when creating template app?

### ISSUE THREAD

**Issue by rpodcast** [ThinkR-open/golem#18]
Due to various "requirements" at the `$day_job` I have to use GitHub repository names that do not fit as a valid R package names to host my app's code (for example `AAA_ZZZ_myappname`).  I want to start using `golem` for these work apps, but when I try to create a new app in the local folder of said repository I am hit with an error when `check_package_name()` is run.  Since I won't be submitting the majority of these apps to CRAN, is there a way we can make the execution of `check_package_name()` in [`create_shiny_template`](https://github.com/ThinkR-open/golem/blob/master/R/create_shiny_template.R) an optional check, instead of mandatory?

**Comment by ColinFay** [comment]
Mmh, I'm quite torn on that request — in a sense, we're trying to promote good practices, so that includes valid package name. Leaving the ability to unknowingly create bad package name is IMO not the way to go.  
On the other hand, it would make sense to be able to do what you are asking for. 

I think the best way to achieve that is not to make it optional, but rather to check if the package name is valid, and if it's not, ask the user with `{yesno}` if they are sure they want to use this name. So in a sense, it's not optional, but conditional to the user answer — that way, they won't unknowingly use an illegal package name, if they say yes, they know that they're doing that.

**Comment by VincentGuyader** [comment]
using {yesno}  would not allow the use of the Rstudio human interface for creating the template...
I think we can use this instead : 
`golem::create_shiny_template(check_name=FALSE)`

If you really want a valid name, use command line and use explictly `check_name=FALSE`

**Comment by ColinFay** [comment]
Not sure what you mean by not allowing the use of the RStudio interface. 

Tried here and asking yesno seems to be working : 

![Screenshot 2019-03-19 at 12 03 33](https://user-images.githubusercontent.com/17936236/54601369-2f642c80-4a3f-11e9-934a-044a3b83c46a.png)
![Screenshot 2019-03-19 at 12 04 31](https://user-images.githubusercontent.com/17936236/54601370-2f642c80-4a3f-11e9-9630-ccf824cfa031.png)

**Comment by VincentGuyader** [comment]
![image](https://user-images.githubusercontent.com/10470699/54602945-53297180-4a43-11e9-8ce2-240c95739d51.png)

**Comment by ColinFay** [comment]
Still works even with the yesno :)
(there is currently another yesno in the package creation function ;) )

Anyway, to get back to the original question, indeed this shouldn't be something allowed without a conscious action of the user, and not something we should promote (don't do it unless you have a very specific reason to do it). 

So the idea of doing `golem::create_shiny_template(path, check_name=FALSE)` seems like the right way to go: the user has to be a little bit knowledgable about what they are doing. 

So, with the widget — no illegal name, and command-line only solution for non-checking package name :) 

How does it sound @rpodcast?

**Comment by rpodcast** [comment]
@ColinFay I like that approach.  It is a "niche" situation so I have no problem using the function directly to bootstrap the app.  I'll be glad to test once that optional flag is added.

**Comment by ColinFay** [comment]
@rpodcast Implemented in #8008fcd :) 

The only "issue" by doing it that way is that you need to manually navigate to the project. But it can easily be done through `rstudioapi::openProject()`

``` r 
> golem::create_shiny_template("../1234", check_name = FALSE)
── Created ───────────────────────────────────────
> rstudioapi::openProject("../1234")
```

Let me know what you think.

**Comment by rpodcast** [comment]
Works perfectly for my needs!  My typical workflow in this situation is to set up an RStudio project (i.e. a generic new project) and then the function inside the project dir with `golem::create_shiny_template(".", check_name = FALSE)`. Appreciate the fix and this will play a key role in using `golem` for my production `$day_job` apps.
